examples-gallery/plot_one_legged_time_trial.py

```diff
@@ -0,0 +1,351 @@
+"""
+Single human leg with four driving lumped muscles. The crank inertia and
+resistance mimic the torque felt if accelerating the whole bicycle with rider
+on flat ground.
+
+The goal will be travel a specific distance in the shortest amount of time
+given that the leg muscles have to coordinate and can only be excited from 0 to
+1.
+
+Second goal will then be to solve for crank length and seat height that gives
+optimal performance.
+
+"""
+import numpy as np
+import sympy as sm
+import sympy.physics.mechanics as me
+import sympy.physics.biomechanics as bm
+import opty
+
+# bike frame, crank, pedal/foot, lower leg, upper leg
+N, A, B, C, D = sm.symbols('N, A, B, C, D', cls=me.ReferenceFrame)
+
+q1, q2, q3, q4 = me.dynamicsymbols('q1, q2, q3, q4')
+u1, u2, u3, u4 = me.dynamicsymbols('u1, u2, u3, u4')
+q = sm.Matrix([q1, q2, q3, q4])
+u = sm.Matrix([u1, u2, u3, u4])
+
+ls, lc, lf, ll, lu = sm.symbols('ls, lc, lf, ll, lu', real=True, positive=True)
+lam, g, rk = sm.symbols('lambda, g, rk', real=True)
+mA, mB, mC, mD = sm.symbols('mA, mB, mC, mD')
+IAzz, IBzz, ICzz, IDzz = sm.symbols('IAzz, IBzz, ICzz, IDzz')
+J, m, rw, G, Cr, CD, rho, Ar = sm.symbols('J, m, rw, G, Cr, CD, rho, Ar')
+
+A.orient_axis(N, N.z, q1)  # crank angle
+B.orient_axis(A, A.z, q2)  # pedal/foot angle
+C.orient_axis(B, B.z, q3)  # ankle angle
+D.orient_axis(C, C.z, q4)  # knee angle
+
+A.set_ang_vel(N, u1*N.z)
+B.set_ang_vel(A, u2*A.z)
+C.set_ang_vel(B, u3*B.z)
+D.set_ang_vel(C, u4*C.z)
+
+# P1 : crank center
+# P2 : pedal center
+# P3 : ankle
+# P4 : knee
+# P5 : hip
+# P6 : seat center
+# P7 : heel
+P1, P2, P3, P4, P5, P6, P7 = sm.symbols('P1, P2, P3, P4, P5, P6, P7',
+                                        cls=me.Point)
+Ao, Bo, Co, Do = sm.symbols('Ao, Bo, Co, Do', cls=me.Point)
+
+Ao.set_pos(P1, 0*A.x)
+P2.set_pos(P1, lc*A.x)  # pedal center
+Bo.set_pos(P2, lf/2*B.x)  # foot mass center
+P3.set_pos(P2, lf*B.x)  # ankle
+P7.set_pos(P2, 3*lf/2*B.x)  # heel
+Co.set_pos(P3, ll/2*C.x)  # lower leg mass center
+P4.set_pos(P3, ll*C.x)  # knee
+Do.set_pos(P4, lu/2*D.x)  # upper leg mass center
+P5.set_pos(P4, lu*D.x)  # hip
+P6.set_pos(P1, -ls*sm.cos(lam)*N.x + ls*sm.sin(lam)*N.y)  # seat
+
+P1.set_vel(N, 0)
+P6.set_vel(N, 0)
+Ao.v2pt_theory(P1, N, A)
+P2.v2pt_theory(P1, N, A)
+P7.v2pt_theory(P2, N, B)
+Bo.v2pt_theory(P2, N, B)
+P3.v2pt_theory(P2, N, B)
+Co.v2pt_theory(P3, N, C)
+P4.v2pt_theory(P3, N, C)
+Do.v2pt_theory(P4, N, D)
+P5.v2pt_theory(P4, N, D)
+
+kindiff = sm.Matrix([ui - qi.diff() for ui, qi in zip(u, q)])
+
+holonomic = (P5.pos_from(P1) - P6.pos_from(P1)).to_matrix(N)[:2, :]
+
+IA = me.Inertia.from_inertia_scalars(Ao, A, 0, 0, IAzz + (J + m*rw**2)*G)
```

Comment:
```suggestion
IA = me.Inertia.from_inertia_scalars(Ao, A, 0, 0, IAzz + (2*J + m*rw**2)*G)
```

account for two wheels